### ISSUE TITLE
Mexc - Unable to use any market pair

### ISSUE THREAD

**Issue by rapcmia** [hummingbot/hummingbot#7174]
### Describe the bug

When checking mexc, we are able to add the connector API successfully and fetch balance ok however when selecting market pair on strategy (v1) it does not seem to be displaying the list fetch from exchange

![image](https://github.com/user-attachments/assets/3fa083c7-9cb9-42a3-a7d7-634206995418)
  - Already added the API keys so it should return or display market pairs on autocomplete
  - Not display anything related to USDT nor USDC
    ![image](https://github.com/user-attachments/assets/4702a5ea-cc5d-4f1a-9530-89e055150ac5)
  - The balance command seems to be working ok 
    ![image](https://github.com/user-attachments/assets/7e0f4d24-6db5-45ab-a23d-66ac191a51ba)
  - Start strategy on v2 using pmm_simple controller and getting the following errors:
    - `Unhandled error in background task..KeyError:...`
      ```
      2024-08-16 18:46:53,419 - 19789 - hummingbot.client.hummingbot_application - INFO - Creating the clock with tick size: 1.0
      2024-08-16 18:46:53,421 - 19789 - hummingbot.client.hummingbot_application - INFO - start command initiated.
      2024-08-16 18:46:54,028 - 19789 - hummingbot.strategy.script_strategy_base - WARNING - mexc is not ready. Please wait...
      2024-08-16 18:46:54,271 - 19789 - hummingbot.connector.exchange.mexc.mexc_exchange.MexcExchange - INFO - Network status has changed to NetworkStatus.CONNECTED. Starting networking...
      2024-08-16 18:46:54,363 - 19789 - hummingbot.core.utils.async_utils - ERROR - Unhandled error in background task: 'TORN-USDT'
      Traceback (most recent call last):
        File "/Users/rapcomia/github/hummingbot/hummingbot/staging/hummingbot/core/utils/async_utils.py", line 9, in safe_wrapper
          return await c
        File "/Users/rapcomia/github/hummingbot/hummingbot/staging/hummingbot/core/data_type/order_book_tracker.py", line 180, in _init_order_books
          self._order_books[trading_pair] = await self._initial_order_book_for_trading_pair(trading_pair)
        File "/Users/rapcomia/github/hummingbot/hummingbot/staging/hummingbot/core/data_type/order_book_tracker.py", line 173, in _initial_order_book_for_trading_pair
          return await self._data_source.get_new_order_book(trading_pair)
        File "/Users/rapcomia/github/hummingbot/hummingbot/staging/hummingbot/core/data_type/order_book_tracker_data_source.py", line 65, in get_new_order_book
          snapshot_msg: OrderBookMessage = await self._order_book_snapshot(trading_pair=trading_pair)
        File "/Users/rapcomia/github/hummingbot/hummingbot/staging/hummingbot/connector/exchange/mexc/mexc_api_order_book_data_source.py", line 113, in _order_book_snapshot
          snapshot: Dict[str, Any] = await self._request_order_book_snapshot(trading_pair)
        File "/Users/rapcomia/github/hummingbot/hummingbot/staging/hummingbot/connector/exchange/mexc/mexc_api_order_book_data_source.py", line 52, in _request_order_book_snapshot
          "symbol": await self._connector.exchange_symbol_associated_to_pair(trading_pair=trading_pair),
        File "hummingbot/connector/exchange_base.pyx", line 97, in exchange_symbol_associated_to_pair
          return symbol_map.inverse[trading_pair]
        File "/Users/rapcomia/miniconda3/envs/hummingbot/lib/python3.10/site-packages/bidict/_base.py", line 524, in __getitem__
          return self._fwdm[key]
      KeyError: 'TORN-USDT'
      ```
    - `Unexpected error occurred subscribing to order book trading and delta streams...`
      ```
      2024-08-16 18:46:54,458 - 19789 - hummingbot.connector.exchange.mexc.mexc_api_user_stream_data_source.MexcAPIUserStreamDataSource - INFO - Successfully obtained listen key 2b3cce36d57af7d262e816190e1b9ae248df3b8f6e9660c0749b551dcbeb365c
      2024-08-16 18:46:54,712 - 19789 - hummingbot.connector.exchange.mexc.mexc_api_user_stream_data_source.MexcAPIUserStreamDataSource - INFO - Subscribed to private order changes and balance updates channels...
      2024-08-16 18:46:54,774 - 19789 - hummingbot.connector.exchange.mexc.mexc_api_order_book_data_source.MexcAPIOrderBookDataSource - ERROR - Unexpected error occurred subscribing to order book trading and delta streams...
      Traceback (most recent call last):
        File "/Users/rapcomia/github/hummingbot/hummingbot/staging/hummingbot/connector/exchange/mexc/mexc_api_order_book_data_source.py", line 76, in _subscribe_channels
          symbol = await self._connector.exchange_symbol_associated_to_pair(trading_pair=trading_pair)
        File "hummingbot/connector/exchange_base.pyx", line 97, in exchange_symbol_associated_to_pair
          return symbol_map.inverse[trading_pair]
        File "/Users/rapcomia/miniconda3/envs/hummingbot/lib/python3.10/site-packages/bidict/_base.py", line 524, in __getitem__
          return self._fwdm[key]
      KeyError: 'TORN-USDT'
      ```
       - This also return the same traceback on the following errors found on the attach log file:
         - `Unexpected error occurred when listening to order book streams. Retrying in 5 seconds...`
         - `Unexpected error while fetching account updates.`

### Steps to reproduce

1.  Add mexc connector, runing `balance` command should display the assets on account if successfully added
2. Setup v1 pure market making, select `mexc` as spot connector and check autocomplete list or add any tokens to use 
3. Setup v2 pmm_simple then start, observe the errors mentioned above

### Release version

development 2.0.1

### Type of installation

Source

### Attach required files

[logs_conf_pmm_mexc1.log](https://github.com/user-attachments/files/16636346/logs_conf_pmm_mexc1.log)

**Comment by Kaiwen0418** [comment]
Got the same bug, tried to remove filtering of valid trading pair and solved. However v1 strategy then froze without even error log

**Comment by Kaiwen0418** [comment]
> Got the same bug, tried to remove filtering of valid trading pair and solved.
> 
> ```
> def  _initialize_trading_pair_symbols_from_exchange_info(self, exchange_info: Dict[str, Any]):
> 
>         mapping = bidict()
> 
>         # for symbol_data in filter(mexc_utils.is_exchange_information_valid, exchange_info["symbols"]):
> 
>         # this filter failded some how
> 
>         for symbol_data in exchange_info["symbols"]:
> ```
> 
> However v1 strategy then froze without even error log

Solved, MEXC changed their exchange info API, the `status` attribute now return `1` instead of `ENABLED`
```
def is_exchange_information_valid(exchange_info: Dict[str, Any]) -> bool:
    """
    Verifies if a trading pair is enabled to operate with based on its exchange information
    :param exchange_info: the exchange information for a trading pair
    :return: True if the trading pair is enabled, False otherwise
    """
    return exchange_info.get("status", None) == "1" and "SPOT" in exchange_info.get("permissions", list()) \
        and exchange_info.get("isSpotTradingAllowed", True) is True
```

**Comment by nikspz** [comment]
Thanks for your participation, @Kaiwen0418. The fix has been merged to latest development branch and will be available in hummingbot version 2.0.1.
